src/Generator/Generators/CSharp/CSharpTypePrinter.cs

```diff
@@ -661,6 +661,10 @@ public override TypePrinterResult VisitParameter(Parameter param, bool hasName)
             var extension = param.Kind == ParameterKind.Extension ? "this " : string.Empty;
             var usage = GetParameterUsage(param.Usage);
 
+            if (!hasName &&
+                (param.DefaultArgument == null || !Options.GenerateDefaultValuesForArguments))
```

Comment:
It's impossible to have a default value without a name anyway, so only leave the check for a name.